### ISSUE TITLE
Pass-through 'System.out & err' into output window during execution of JUnit test

### ISSUE THREAD

**Issue by xtracoder** [kelemen/netbeans-gradle-project#276]
While developing my code and testing it right-away via unit tests I'd like to step through my test method and at some point print something diagnostic to System.out and **immediately** see it somewhere, most probably in output window.

Currently all output is accumulated somewhere within JUnit test runnner and displayed in Test Results only after tests are finished. This is really not convenient.

This functionality is working as expected for MAVEN project.

After some investigation it appears that major part of problem comes from Gradle's paradigm of minimizing the console output and can be easily fixed via following [gradle configuration](https://docs.gradle.org/current/dsl/org.gradle.api.tasks.testing.Test.html) (look for showStandardStreams) ...

```
test.testLogging { 
    showStandardStreams = true
} 
```
... which I think is really not needed for regular build, but is helpful when running test from IDE - it makes sense to add an options for this in Gradle runner.

But ... after setting  `showStandardStreams = true` output appears when starting gradle from command line, but is still stuck in IDE - it does appear in output __after__ test is completed, not right-away. I have no idea where this unwanted buffering is performed. Needs to be fixed.

Test projects (Maven | Gradle) attached.
[junit-output-test.zip](https://github.com/kelemen/netbeans-gradle-project/files/270350/junit-output-test.zip)

**Comment by xtracoder** [comment]
It appears that 'buffering' does appear when stepping through code via `Debug->Step Over`
```
        System.out.println("1");
        System.out.println("2");
        System.out.println("3");
```

But messages appear immediately when code is running in un-paused manner, for example this code after `Debug->Continue`
```        
        System.out.println("" + new Date() + " - started");
        for( int i = 0; i < 100; i++ ) {
            System.out.println("" + new Date() + " - continue");
            Thread.sleep(500);
        }
```

**Comment by kelemen** [comment]
I cannot completely reproduce what you are saying. That is, writing `test.testLogging.showStandardStreams = true` will not do any buffering for me but immediately outputs it.

If you just want to set this property only when running a task from NetBeans, you can apply an init script which sets it. That is, you can add an *--init-script* argument in the global settings (Options/Misc/Gradle/Script&Tasks, Gradle Arguments) which sets it.

**Comment by xtracoder** [comment]
And after some debugging it looks like output is not 'buffered' - it is 'suspended'. The thread which takes output from Gradle and forwards it to NetBeans is sleeping in the Step by Step debugging and the problem appears only when initial breakpoint which interrupted execution is set to 'Suspend = All Threads'. After changing it to 'Suspend = Breakpoint thread' output during stepping through code is emitted immediately. So ... since same works correctly for Maven there should be some way to **not** suspend the thread with the output handler.

**Comment by kelemen** [comment]
I think this is a generic non-deterministic bug in NetBeans (or in whatever NB uses, actually I have seen this issue in Eclipse as well). Sometimes, NB fails to restart threads after doing some debugging steps. This is not deterministic. I have tried what you wrote and if I single stepped through the print lines, I got the output in the output window immediately.

So, I believe this is not an issue with this plugin.

**Comment by kelemen** [comment]
Since, I cannot do anything with this issue, I'm closing it. Reopen or open a new issue the issue is reproducible and is caused by this plugin.

**Comment by wadechandler** [comment]
If logging is processed by another thread, as I believe it is, then normal breakpoint operation will do what you see. Google has many examples, but this may be useful. http://stackoverflow.com/questions/8174408/does-a-breakpoint-halt-all-threads
